code/datums/martial/sleeping_carp.dm

```diff
@@ -148,6 +146,23 @@
 		return TRUE
 	return ..()
 
+/datum/martial_art/the_sleeping_carp/on_projectile_hit(mob/living/carbon/human/A, obj/item/projectile/P, def_zone)
+	. = ..()
+	if(A.incapacitated(FALSE, TRUE)) //NO STUN
+		return
+	if(!(A.mobility_flags & MOBILITY_USE)) //NO UNABLE TO USE
+		return
+	if(A.dna && A.dna.check_mutation(HULK)) //NO HULK
+		return
+	if(!isturf(A.loc)) //NO MOTHERFLIPPIN MECHS!
+		return
```

Comment:
Are these supposed to be returning null rather than `BULLET_ACT_HIT`?

Reply:
They return what the parent returns. Which is BULLET_ACT_HIT
```
/datum/martial_art/proc/on_projectile_hit(mob/living/carbon/human/A, obj/item/projectile/P, def_zone)
	return BULLET_ACT_HIT

Reply:
Guess I could also add that there. Wouldn't change any functionality, but does look a bit better.

Reply:
Ah, probably fine then